tools/fuzzer.cpp

```diff
@@ -0,0 +1,403 @@
+            << std::endl;
+  std::cout << "Available format options are:" << std::endl;
+  std::cout << " --vpack       create VPack."
+            << std::endl;
+  std::cout << " --json        create JSON."
+            << std::endl;
+  std::cout << " --i <number>  number of iterations (number > 0). Default: 1"
+            << std::endl;
+  std::cout << " --t <number>  number of threads (number > 0). Default: 1"
+            << std::endl;
+  std::cout
+      << " --s <number> number that will be used as seed for random generation (number >= 0). Default: random_device"
+      << std::endl;
+}
+
+static inline bool isOption(char const* arg, char const* expected) {
+  return (strcmp(arg, expected) == 0);
+}
+
+uint32_t randWithinRange(uint32_t min, uint32_t max, RandomGenerator& randomGenerator) {
+  std::uniform_int_distribution<> distr(min, max);
+  return distr(randomGenerator.mt64);
+}
+
+void appendRandUtf8Char(RandomGenerator& randomGenerator, std::string& utf8Str) {
+  using limits = KnownLimitValues;
+  int numBytes = randWithinRange(1, 4, randomGenerator);
+  std::vector<char> utf8CharBytes;
+  utf8CharBytes.reserve(numBytes);
+  switch (numBytes) {
+    case 1: {
+      utf8CharBytes.emplace_back(
+          randWithinRange(limits::utf81ByteFirstLowerBound, limits::utf81ByteFirstUpperBound, randomGenerator));
+      break;
+    }
+    case 2: {
+      utf8CharBytes.emplace_back(randWithinRange(limits::utf82BytesFirstLowerBound, limits::utf82BytesFirstUpperBound,
+                                                 randomGenerator));
+      utf8CharBytes.emplace_back(
+          randWithinRange(limits::utf8CommonLowerBound, limits::utf8CommonUpperBound, randomGenerator));
+      break;
+    }
+    case 3: {
+      utf8CharBytes.emplace_back(randWithinRange(limits::utf83BytesFirstLowerBound, limits::utf83BytesFirstUpperBound,
+                                                 randomGenerator));
+      for (uint32_t i = 0; i < 2; ++i) {
+        utf8CharBytes.emplace_back(
+            randWithinRange(limits::utf8CommonLowerBound, limits::utf8CommonUpperBound, randomGenerator));
+      }
+      break;
+    }
+    case 4: {
+      utf8CharBytes.emplace_back(randWithinRange(limits::utf84BytesFirstLowerBound, limits::utf84BytesFirstUpperBound,
+                                                 randomGenerator));
+      for (uint32_t i = 0; i < 3; ++i) {
+        utf8CharBytes.emplace_back(
+            randWithinRange(limits::utf8CommonLowerBound, limits::utf8CommonUpperBound, randomGenerator));
+      }
+      break;
+    }
+    default:
+      assert(false);
+  }
+  utf8CharBytes.emplace_back('\0');
+  utf8Str.append(std::string(utf8CharBytes.data()));
+}
+
+static void generateUtf8String(RandomGenerator& randomGenerator, std::string& utf8Str) {
+  using limits = KnownLimitValues;
+  std::uniform_int_distribution<> distr(limits::minUtf8RandStringLength, limits::maxUtf8RandStringLength);
```

Comment:
Why does this use its own distribution? Works, but why not go with the modulo trick that is used in all other places?